app_base.htm

```diff
@@ -0,0 +1,79 @@
+<%inherit file="homepage/templates/base.htm" />
+<%! from catalog import models as cmod %>
+
+##set the active tab dynamically with code
+<%block name="menu">
+  <li class="${'active' if request.dmp.page == 'index' else ''}"><a href="/homepage/index/">Home</a></li>
+  <li class="${'active' if request.dmp.page == 'about' else ''}"><a href="/homepage/about/">About</a></li>
+  <li class="${'active' if request.dmp.page == 'contact' else ''}"><a href="/homepage/contact/">Contact</a></li>
+  <li class="${'active' if request.dmp.page == 'terms' else ''}"><a href="/homepage/terms/">Terms</a></li>
+  <li class="${'active' if request.dmp.page == 'faq' else ''}"><a href="/homepage/faq/">FAQ</a></li>
+  <li class="${'active' if request.dmp.page == 'product' else ''}"><a href="/catalog/product/">Products</a></li>
+</%block>
+
+<%block name="login">
+  % if request.user.is_authenticated:
+  <li class="dropdown">
+    <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+      <span class="glyphicon glyphicon-log-in>"></span>
+      ${ request.user }
+      <span class="caret"></span>
+    </a>
+    <ul class="dropdown-menu">
+      <li><a href="/account/logout/?next=/index/">Logout</a></li>
+      <li><a href="/admin/">View Account</a></li>
+    </ul>
+  </li>
+  % else:
+  <li>
+    <a href="/account/signup/">Sign Up</a>
+  </li>
+  <li><a href="/account/login/">Login</a></li>
+  % endif
+</%block>
+
+<%block name="content_left">
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+<div class="sidebar">
```

Comment:
naming class 'leftbar' would be more clear since you have a 'rightbar' in the content_right block

Reply:
Wow I didn't even notice! 😲 I 100% agree.